From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lee Jones <lee.jones@linaro.org>
Date: Wed, 12 May 2021 21:22:02 +0100
Subject: Revert "fuse: fix matching of FUSE_DEV_IOC_CLONE command"

This reverts commit 6076f5f341e612152879bfda99f0b76c1953bf0b.

This is currently breaking avd/avd_boot_test_kernel_triggered on
kernel_virt_x86_64, when tested against android-mainline which is
preventing an up-level to v5.13-rc1.  We can do some more
investigation later, but for now, we'll simply revert the change.

Signed-off-by: Lee Jones <lee.jones@linaro.org>
Change-Id: I56d3360a7cc0f176da102ab59fdcf2e3b9f2a46d
---
 fs/fuse/dev.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/fs/fuse/dev.c b/fs/fuse/dev.c
--- a/fs/fuse/dev.c
+++ b/fs/fuse/dev.c
@@ -2233,8 +2233,11 @@ static long fuse_dev_ioctl(struct file *file, unsigned int cmd,
 	int oldfd;
 	struct fuse_dev *fud = NULL;
 
-	switch (cmd) {
-	case FUSE_DEV_IOC_CLONE:
+	if (_IOC_TYPE(cmd) != FUSE_DEV_IOC_MAGIC)
+		return -ENOTTY;
+
+	switch (_IOC_NR(cmd)) {
+	case _IOC_NR(FUSE_DEV_IOC_CLONE):
 		res = -EFAULT;
 		if (!get_user(oldfd, (__u32 __user *)arg)) {
 			struct file *old = fget(oldfd);
